apps/dashboard/app/helpers/application_helper.rb

```diff
@@ -82,4 +82,9 @@ def profile_link(profile_info)
     profile_id = profile_info[:id]
     nav_link(profile_info.fetch(:name, profile_id), profile_info.fetch(:icon, "user"), settings_path("settings[profile]" => profile_id), method: "post") if profile_id
   end
+
+  def custom_css_paths
+    public_url = @user_configuration.public_url
+    @user_configuration.custom_css_files.reject{|css| css.blank?}.map { |css_file| File.join(public_url, css_file) }
```

Comment:
```suggestion
    @user_configuration.custom_css_files.map do |css_file| 
      css_file.to_s.empty? ? nil : File.join(@user_configuration.public_url, css_file)
    end.compact.select do |css_path|
      css_path.to_s.start_with?('/')   # do not load CSS files from another host. only allow relative paths.
    end
```


Turns out with the `nonce` you could totally nonce a file that doesn't come from `self`, so I think I would like a check here to be sure everything is served from this host.

Of course it's an admin who sets `public_uri`, but given that this is on _every page_, it gives a bad actor a lot of opportunity.

Reply:
I have added these changes. As well, I have updated the tests to check only relative URLs are allowed.

I have added a new integration test to check that custom CSS files are added to the homepage. Not sure if this test is too much.

Reply:
A comment from Bill after sharing this update:
Should we add a check in UserConfiguration.public_url so that it is always relative? in case other assets are loaded this way?

Reply:
Yea that'd likely be better for `UserConfiguration.public_url` to encapsulate any checks/casts.

Reply:
Added a check into `UserConfiguration.public_url`, what do you think?

Reply:
Yep that works for me!